main.tf

```diff
@@ -17,8 +17,8 @@ resource "aws_sqs_queue" "this" {
   deduplication_scope         = var.deduplication_scope
   fifo_throughput_limit       = var.fifo_throughput_limit
 
-  sqs_managed_sse_enabled           = var.sqs_managed_sse_enabled ? var.sqs_managed_sse_enabled : null
-  kms_master_key_id                 = var.sqs_managed_sse_enabled ? null : var.kms_master_key_id
+  sqs_managed_sse_enabled           = var.kms_master_key_id == null ? var.sqs_managed_sse_enabled : null
```

Comment:
I believe these should be:

```hcl
  sqs_managed_sse_enabled           = var.sqs_managed_sse_enabled
  kms_master_key_id                 = var.sqs_managed_sse_enabled ? null : var.kms_master_key_id
  kms_data_key_reuse_period_seconds = var.sqs_managed_sse_enabled ? null : var.kms_data_key_reuse_period_seconds
```

and then the default value in the `kms_data_key_reuse_period_seconds` variable should be changed to `null` since this is set at the API/provider

Reply:
This doesn't seem to work since we can't have `sqs_managed_sse_enabled` even if set to `false` with `kms_master_key_id` set. 
<img width="1054" alt="Screenshot 2022-12-04 at 11 17 38 AM" src="https://user-images.githubusercontent.com/69476188/205503095-bb9e8067-7d35-4cbe-8bdc-06881e480ffb.png">